src/object_store/src/object/s3.rs

```diff
@@ -100,6 +90,32 @@ impl ObjectStore for S3ObjectStore {
         })
     }
 
+    async fn streaming_read(
```

Comment:
We don't run tests here. We should run e2e tests for S3 once we have the `SSTableStreamIterator` implemented.